src/track/external-captions-handler.js

```diff
@@ -0,0 +1,397 @@
+      if (this._player.config.playback.useNativeTextTrack && event.payload.selectedTextTrack.external) {
+        this.selectTextTrack(event.payload.selectedTextTrack);
+      } else if (!this._player.config.playback.useNativeTextTrack && !event.payload.selectedTextTrack.external) {
+        this._eventManager.unlisten(this._player, Html5EventType.TIME_UPDATE);
+        this._player.onCueChange({
+          payload: {
+            cues: []
+          }
+        });
+      }
+    }
+  }
+
+  /**
+   * setting and adding a text track to the video element
+   * @param {TextTrack } textTrack - the player module TextTrack that will be set
+   * @returns {TextTrack} - the DOM text track that was set
+   * @private
+   */
+  _setNativeTextTrack(textTrack: TextTrack): TextTrack {
+    let _domTrack;
+    const _videoElement = this._player.getVideoElement();
+    const _sameLanguageTrackIndex = this._indexOfSameLanguageTrack(textTrack);
+    if (_sameLanguageTrackIndex > -1) {
+      _domTrack = _videoElement.textTracks[_sameLanguageTrackIndex];
+      _domTrack.mode = 'showing';
+      if (_domTrack.cues) {
+        Object.values(_domTrack.cues).forEach(cue => {
+          _domTrack.removeCue(cue);
+        });
+      }
+    } else {
+      _domTrack = _videoElement.addTextTrack("subtitles", textTrack.label, textTrack.language);
+      this._player._getTracksByType(TrackType.TEXT).map(track => {
+        track.index = track.index + 1;
+      });
+      textTrack.index = parseInt(Object.keys(_videoElement.textTracks).find(key => _videoElement.textTracks[key] === _domTrack));
+    }
+    return _domTrack;
+  }
+
+  /**
+   * adding cues to a native text track
+   * @param {TextTrack} textTrack - the text track containing the cues to add
+   * @returns {void}
+   * @private
+   */
+  _addCuesToNativeTrack(textTrack: TextTrack): void {
+    const track = Object.values(this._player.getVideoElement().textTracks).filter(track => track.language === textTrack.language)[0];
+    textTrack.cues.forEach(cue => {
+      track.addCue(cue);
+    });
+  }
+
+  /**
+   * setting an external track as active, de-actvating all the rest and start listening to time updates.
+   * @param {TextTrack} textTrack - text track to be set
+   * @private
+   */
+  _setTextTrack(textTrack: textTrack): void {
+    if (!this._player.config.playback.useNativeTextTrack) {
+      const track = this._player._getTracksByType(TrackType.TEXT).filter(track => {
+        if (track.index !== textTrack.index) {
+          track.active = false;
+        } else {
+          return true;
+        }
+      })[0];
+      if (!track.active) {
+        this._player.disableAllTextTracks();
+        ExternalCaptionsHandler._logger.debug('External text track changed', track);
+        track.active = true;
+        this._player.dispatchEvent(new FakeEvent(CustomEventType.TEXT_TRACK_CHANGED, {selectedTextTrack: track}));
+        this._handleExternalCaptionsCallback = () => {
```

Comment:
I think you don't need the bounded method, cause the eventManager takes care of it, just pass it the handler and remove it where you need to